src/HotChocolate/Core/src/Execution/Processing/VariableCoercionHelper.cs

```diff
@@ -142,10 +142,11 @@ public VariableCoercionHelper(InputFormatter inputFormatter, InputParser inputPa
                 case ListValueNode lv:
                     return Rewrite(inputType, lv);
 
-                case StringValueNode sv:
-                    return inputType.Kind is TypeKind.Enum
-                        ? new EnumValueNode(sv.Location, sv.Value)
-                        : node;
+                case StringValueNode sv when inputType.IsEnumType():
```

Comment:
I have simplified this to the IsEnumType since I have optimized the implementation of IsEnumType to check the kind instead of the type.